### PR TITLE
[SPARK-10277][MLlib][PySpark] Add @since annotation to pyspark.mllib.regression

### PR DESCRIPTION
**Pull request by yu-iskw**

### CODE REVIEW

File: python/pyspark/mllib/regression.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ class LabeledPoint(object):
             column matrix)
 
     Note: 'label' and 'features' are accessible as class attributes.
+
+    .. addedversion:: 1.0.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 2]
```diff
@@ -65,6 +67,8 @@ class LinearModel(object):
 
     :param weights: Weights computed for every feature.
     :param intercept: Intercept computed for this model.
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 3]
```diff
@@ -93,8 +101,11 @@ class LinearRegressionModelBase(LinearModel):
     True
     >>> abs(lrmb.predict(SparseVector(2, {0: -1.03, 1: 7.777})) - 14.624) < 1e-6
     True
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 4]
```diff
@@ -162,14 +173,20 @@ class LinearRegressionModel(LinearRegressionModelBase):
     True
     >>> abs(lrm.predict(SparseVector(1, {0: 1.0})) - 1) < 0.5
     True
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 5]
```diff
@@ -198,8 +215,20 @@ def _regression_train_wrapper(train_func, modelClass, data, initial_weights):
...
+    """
+    Train a linear regression model with no regularization using Stochastic Gradient Descent.
+    This solves the least squares regression formulation
+                 f(weights) = 1/n ||A weights-y||^2^
+    (which is the mean squared error).
+    Here the data matrix has n rows, and the input RDD holds the set of rows of A, each with
+    its corresponding right hand side label y.
+    See also the documentation for the precise formulation.
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 6]
```diff
@@ -309,14 +338,20 @@ class LassoModel(LinearRegressionModelBase):
     True
     >>> abs(lrm.predict(SparseVector(1, {0: 1.0})) - 1) < 0.5
     True
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 7]
```diff
@@ -326,8 +361,19 @@ def load(cls, sc, path):
...
 class LassoWithSGD(object):
+    """
+    Train a regression model with L1-regularization using Stochastic Gradient Descent.
+    This solves the l1-regularized least squares regression formulation
+             f(weights) = 1/2n ||A weights-y||^2^  + regParam ||weights||_1
+    Here the data matrix has n rows, and the input RDD holds the set of rows of A, each with
+    its corresponding right hand side label y.
+    See also the documentation for the precise formulation.
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 8]
```diff
@@ -428,14 +474,20 @@ class RidgeRegressionModel(LinearRegressionModelBase):
     True
     >>> abs(lrm.predict(SparseVector(1, {0: 1.0})) - 1) < 0.5
     True
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 9]
```diff
@@ -445,8 +497,19 @@ def load(cls, sc, path):
...
 class RidgeRegressionWithSGD(object):
+    """
+    Train a regression model with L2-regularization using Stochastic Gradient Descent.
+    This solves the l2-regularized least squares regression formulation
+             f(weights) = 1/2n ||A weights-y||^2^  + regParam/2 ||weights||^2^
+    Here the data matrix has n rows, and the input RDD holds the set of rows of A, each with
+    its corresponding right hand side label y.
+    See also the documentation for the precise formulation.
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 10]
```diff
@@ -523,6 +586,8 @@ class IsotonicRegressionModel(Saveable, Loader):
     ...     rmtree(path)
     ... except OSError:
     ...     pass
+
+    .. addedversion:: 1.4.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 11]
```diff
@@ -571,8 +641,29 @@ def load(cls, sc, path):
...
+
+    Sequential PAV parallelization based on:
+    Kearsley, Anthony J., Richard A. Tapia, and Michael W. Trosset.
+      "An approach to parallelizing isotonic regression."
+      Applied Mathematics and Parallel Computing. Physica-Verlag HD, 1996. 141-147.
+      Available from [[http://softlib.rice.edu/pub/CRPC-TRs/reports/CRPC-TR96640.pdf]]
+
+    @see [[http://en.wikipedia.org/wiki/Isotonic_regression Isotonic regression (Wikipedia)]]
+
+    .. addedversion:: 1.4.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 12]
```diff
@@ -590,10 +681,13 @@ class StreamingLinearAlgorithm(object):
     Base class that has to be inherited by any StreamingLinearAlgorithm.
 
     Prevents reimplementation of methods predictOn and predictOnValues.
+
+    .. addedversion:: 1.5.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

File: python/pyspark/mllib/regression.py [hunk 13]
```diff
@@ -640,6 +736,8 @@ class StreamingLinearRegressionWithSGD(StreamingLinearAlgorithm):
     :param: numIterations Total number of iterations run.
     :param: miniBatchFraction Fraction of data on which SGD is run for each
                               iteration.
+
+    .. addedversion:: 1.5.0
```
Comment (noel-smith) [inline python/pyspark/mllib/regression.py]: Switch to `.. versionadded`

**Comment by noel-smith** [comment]
Looks good - just need to replace `addedversion` with `versionadded` in the class docstrings.

**Comment by noel-smith** [comment]
That's great - LGTM!

**Comment by mengxr** [comment]
@yu-iskw Could you merge master? There are some merge conflicts.

**Comment by yu-iskw** [comment]
All right. I have rebased this PR with the master branch.

**Comment by mengxr** [comment]
@yu-iskw Please keep the content minimal. This PR shouldn't contain doc changes. Otherwise, it would slow down the code review process. For example, there are issues with docstring line width, links, and other formatting issues. You can see them by running `make html` under `python/docs`. There are also issues with generated HTML doc. So let's remove the doc changes from this PR, and address them in a follow-up PR. Please create a JIRA for it. Thanks!

**Comment by yu-iskw** [comment]
@mengxr as you suggested, it would be great not to contain doc changes in this PR. However,  `@since` tag needs any comment. Otherwise, it doesn't work. Should we add `@since` tag, after adding comments to methods which don't have any comments?

**Comment by noel-smith** [comment]
I think we should keep the comments - but just tweak them so that they're valid [reStructuredText](http://sphinx-doc.org/rest.html). 

In this case all that really needs to be done is to remove the square brackets from URLs and format the indented blocks in the same way as the formula in the ```RidgeRegressionWithSGD.train()``` method (i.e. 4 lines indent + newlines before + after).

**Comment by mengxr** [comment]
@noel-smith @yu-iskw Mixing multiple features in a PR generally delays the code review. For example, we are now discussing `reStructuredText` format for comments in a PR titled "Add @since annotation to pyspark.mllib.regression". I'm going to merge this. @yu-iskw Could you make a follow-up PR to fix the syntax in the comments? Also pay attention to the line width in docstring.

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**